src/promptflow/promptflow/_sdk/_submitter/experiment_orchestrator.py

```diff
@@ -2,23 +2,39 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 import argparse
+import copy
 import hashlib
 import json
 import os
 import platform
 import signal
 import subprocess
 import sys
+import tempfile
 from concurrent import futures
 from concurrent.futures import ThreadPoolExecutor
 from datetime import datetime
 from os import PathLike
 from pathlib import Path
+from typing import Union
+
 import psutil
 
-from promptflow._sdk._constants import ExperimentNodeRunStatus, ExperimentNodeType, ExperimentStatus, FlowRunProperties
+if sys.stdout is None:
```

Comment:
why this?

Reply:
Add the comments. For the process started in detach mode, stdout/std error will be none. To avoid exception to stdout/stderr calls in the dependency package, point stdout/stderr to devnull.

Reply:
but shall we do this in inport time? Or some place in the entry of detach mode?